src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Byte.cs

```diff
@@ -2011,5 +2011,59 @@ private static bool TryFindFirstMatchedLane(Vector128<byte> mask, Vector128<byte
             matchedLane = BitOperations.TrailingZeroCount(selectedLanes) >> 2;
             return true;
         }
+
+        public static void ReverseByteRef(ref byte buf, nint length)
+        {
+            ref byte first = ref buf;
+            ref byte last = ref Unsafe.Add(ref Unsafe.Add(ref first, length), -1);
+            int numBytesWritten = 0;
+            if (Avx2.IsSupported && Vector256<byte>.Count * 2 <= length)
```

Comment:
The Avx2.IsSupported I think can now be Vector256.IsHardwareAccelerated per the other PR today.

Reply:
I put the `Avx2.IsSupported` guard here because inside this block I use the `Avx2.Shuffle` intrinsic. I think if it were just `Vector256.IsSupported`, you could potentially run an illegal instruction if AVX2 isn't supported on the machine.